src/hash.cr

```diff
@@ -845,6 +876,14 @@ class Hash(K, V)
     raise "Hash table too big"
   end
 
+  private def dig_deep(hash_keys)
```

Comment:
See the `has_key?` method. It references another private method that does more work. My reasoning is that the `deep_dig` method would be the workhorse but the other `dig` and `dig?` methods would be the public facing methods, making the code more DRY.